test/inspector/test-async-hook-teardown-at-debug-end.js

```diff
@@ -1,33 +0,0 @@
-'use strict';
-const common = require('../common');
-common.skipIfInspectorDisabled();
-common.skipIf32Bits();
-
-const spawn = require('child_process').spawn;
-
-const script = `
-const assert = require('assert');
-
-// Verify that inspector-async-hook is registered
-// by checking that emitInit with invalid arguments
-// throw an error.
-// See test/async-hooks/test-emit-init.js
-assert.throws(
-  () => async_hooks.emitInit(),
-  'inspector async hook should have been enabled initially');
-
-process._debugEnd();
-
-// Verify that inspector-async-hook is no longer registered,
-// thus emitInit() ignores invalid arguments
-// See test/async-hooks/test-emit-init.js
```

Comment:
It makes me wonder - are DevTools clients disabling stack trace inspection before they disconnect? What happens when a 3rd-party DevTools client is implemented "incorrectly" and does not call `setAsyncCallStackDepth` before closing the session - will we keep async stack traces enabled in such case, possibly forever if no other DevTool clients ever connects? Are we ok with that?

Reply:
This is a good point. I think the alternative PR https://github.com/nodejs/node/pull/16308 addresses this.